UnityGLTF/Assets/UnityGLTF/Scripts/Cache/RefCountedCacheData.cs

```diff
@@ -80,28 +93,32 @@ private void DestroyCachedData()
 			{
 				for (int j = 0; j < MeshCache[i].Length; j++)
 				{
-					if (MeshCache[i][j] != null)
-					{
-						MeshCache[i][j].Unload();
-					}
+					MeshCache[i][j]?.Dispose();
+					MeshCache[i][j] = null;
 				}
 			}
 
 			// Destroy the cached textures
 			for (int i = 0; i < TextureCache.Length; i++)
 			{
-				if (TextureCache[i] != null)
-				{
-					TextureCache[i].Unload();
-				}
+				TextureCache[i]?.Dispose();
+				TextureCache[i] = null;
 			}
 
 			// Destroy the cached materials
 			for (int i = 0; i < MaterialCache.Length; i++)
 			{
-				if (MaterialCache[i] != null)
+				MaterialCache[i]?.Dispose();
+				MaterialCache[i] = null;
+			}
+
+			// Destroy the cached textures
```

Comment:
nit: images, not textures

```suggestion
			// Destroy the cached images
```